File: trunk/OscarDotNet/component/db/Class_db_milestones.pas
unit Class_db_milestones;

interface

uses
  Class_db,
  Class_db_trail,
  system.collections;

type
  TClass_db_milestones = class(TClass_db)
  private
    db_trail: TClass_db_trail;
  public
    constructor Create;
    procedure Check
      (
      code: cardinal;
      var be_processed: boolean;
      var value: datetime
      );
    procedure MarkProcessed(code: cardinal);
  end;

implementation

uses
  mysql.data.mysqlclient;

constructor TClass_db_milestones.Create;
begin
  inherited Create;
  // TODO: Add any constructor code here
  db_trail := TClass_db_trail.Create;
end;

procedure TClass_db_milestones.Check
  (
  code: cardinal;
  var be_processed: boolean;
  var value: datetime
  );
var
  dr: mysqldatareader;
begin
  self.Open;
  dr := mysqlcommand.Create
    (
    'select be_processed,value'
    + ' from fy_calendar'
    + ' where fiscal_year_id = ' //+ biz_fiscal_years.IdOfCurrent
    +   ' and milestone_code = ' + code.tostring,
    connection
    )
    .ExecuteReader;
  dr.Read;
  be_processed := (dr['be_processed'].tostring = '1');
  value := datetime.Parse(dr['value'].tostring);
  dr.Close;
  self.Close;
end;

procedure TClass_db_milestones.MarkProcessed(code: cardinal);
var
  cmdText: string;
begin
  cmdText := 'update fy_calendar'
    + ' set be_processed = TRUE'
    + ' where fiscal_year_id = ' //+ biz_fiscal_years.IdOfCurrent
    +   ' and milestone_code = ' + code.tostring;
  self.Open;
  mysqlcommand.Create
    (
    db_trail.Saved(cmdText),
    connection
    )
    .ExecuteNonQuery;
  self.Close;
end;

end.

File: trunk/OscarDotNet/component/db/Class_db_milestones.cs
using MySql.Data.MySqlClient;
using System;

using Class_db;
using Class_db_trail;
namespace Class_db_milestones
{
    public class TClass_db_milestones: TClass_db
    {
        private TClass_db_trail db_trail = null;
        //Constructor  Create()
        public TClass_db_milestones() : base()
        {
            // TODO: Add any constructor code here
            db_trail = new TClass_db_trail();
        }
        public void Check(uint code, out bool be_processed, out DateTime value)
        {
            MySqlDataReader dr;
            this.Open();
            // + biz_fiscal_years.IdOfCurrent
            dr = new MySqlCommand("select be_processed,value" + " from fy_calendar" + " where fiscal_year_id = " + " and milestone_code = " + code.ToString(), this.connection).ExecuteReader();
            dr.Read();
            be_processed = (dr["be_processed"].ToString() == "1");
            value = DateTime.Parse(dr["value"].ToString());
            dr.Close();
            this.Close();
        }

        public void MarkProcessed(uint code)
        {
            string cmdText;
            // + biz_fiscal_years.IdOfCurrent
            cmdText = "update fy_calendar" + " set be_processed = TRUE" + " where fiscal_year_id = " + " and milestone_code = " + code.ToString();
            this.Open();
            new MySqlCommand(db_trail.Saved(cmdText), this.connection).ExecuteNonQuery();
            this.Close();
        }

    } // end TClass_db_milestones

}
